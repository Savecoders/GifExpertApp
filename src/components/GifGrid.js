import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {
	const { data: images, loading } = useFetchGifs(category);

	return (
		<>
			<h3 className='mg-3vw animate__animated animate__fadeIn'>
				<i class='bx bxs-right-arrow bx-xs' /> {category}
			</h3>
			{loading && (
				<p className='animate__animated animate__flash'>
					<i className='loading bx bx-loader-alt bx-spin'></i>
				</p>
			)}
			<div className='card-grid '>
				{images.map((img) => (
					<GifGridItem key={img.id} {...img} />
				))}
			</div>
		</>
	);
};

export default GifGrid;
