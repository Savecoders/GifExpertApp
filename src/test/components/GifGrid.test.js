import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');
describe('test in <GifGrid/>', () => {
	test('shallow GifGrid', () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true,
		});
		const wrapper = shallow(<GifGrid category={'Hunter X Hunter'} />);
		expect(wrapper).toMatchSnapshot();
	});
	test('shallow items and render images', () => {
		const gifs = [
			{
				id: 'ABC',
				url: 'https://localhost/xddddddd/cosa.jpg',
				title: 'cosas de la vida',
			},
			{
				id: 'asdasd',
				url: 'https://localhost/xddddddd/cosa.jpg',
				title: 'cosas de la vida',
			},
			{
				id: 'aaaaaa',
				url: 'https://localhost/xddddddd/cosa.jpg',
				title: 'cosas de la vida',
			},
		];
		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});
		const wrapper = shallow(<GifGrid category={'Hunter X Hunter'} />);

		/* expect(wrapper).toMatchSnapshot(); */
		expect(wrapper.find('p').exists()).toBe(false);
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
	});
});
