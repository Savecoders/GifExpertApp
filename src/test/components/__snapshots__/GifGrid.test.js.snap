// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <GifGrid/> shallow GifGrid 1`] = `
<Fragment>
  <h3
    className="mg-3vw animate__animated animate__fadeIn"
  >
    <i
      class="bx bxs-right-arrow bx-xs"
    />
     
    Hunter X Hunter
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    <i
      className="loading bx bx-loader-alt bx-spin"
    />
  </p>
  <div
    className="card-grid "
  />
</Fragment>
`;
